play-json-tools/src/main/scala/com/evolutiongaming/playjsontools/PlayJsonHelper.scala

```diff
@@ -393,6 +393,52 @@ object PlayJsonHelper {
     }
   }
 
+  object DiscriminatedEitherFormat {
+    def valueFromOnlyKey[A](key: String, fromKeys: Set[String])(implicit ra: Reads[A]): Reads[A] =
+      new Reads[A] {
+        def reads(js: JsValue) = js match {
+          case o: JsObject => o.keys match {
+            case a if (a & fromKeys) == Set(key) => ra.reads(o.values.head)
```

Comment:
why not just pass opposite key and verify that only single of two exist in map without unnecessary manipulations with Set ?

Reply:
uglier, but faster. I can change it